base/sparse/linalg.jl

```diff
@@ -532,11 +531,14 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
     end
     ind = Array(Int64, n)
     ind_hist = Array(Int64, maxiter * t)
-    S = zeros(T, n, t)
+
+    Ti = typeof(float(zero(T)))
```

Comment:
I understand now. We probably don't have any cases where it would differ from `float`, but I guess `typeof(sign(zero(T)))` is the precise type, right?